atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -399,15 +399,16 @@ public void addPool(InetSocketAddress server) {
 
     public void removePool(InetSocketAddress removedServerAddress) {
         blacklist.remove(removedServerAddress);
+        CassandraClientPoolingContainer containerToRemove = currentPools.get(removedServerAddress);
+        currentPools.remove(removedServerAddress);
         try {
-            currentPools.get(removedServerAddress).shutdownPooling();
+            containerToRemove.shutdownPooling();
```

Comment:
technically doesn't fully solve the issue: what if there are in-flight requests on the pool-to-be-closed?

Reply:
hmm, yes - ideally we'd have something that will:
a) block new requests from being started
b) wait for existing requests to complete (or time out)
c) clean up by shutting down pooling.

Reply:
Looks like once we've called `borrowObject` (in `CassandraClientPoolingContainer.runWithGoodResource`), we should be good?

I'm not averse to adding something like this if it's proven to be necessary (and I can follow up this PR with logging to help us to determine this), but for now I think in-flight requests should continue to work.

Reply:
That's pretty nice, actually!